src/request/path_param.rs

```diff
@@ -1,50 +1,45 @@
 use http::StatusCode;
-use std::{collections::HashMap, str::FromStr};
+use std::str::FromStr;
 
 use crate::{Error, Result};
 
-pub type PathParams = HashMap<&'static str, (usize, usize)>;
+pub(crate) type PathParams = Vec<(&'static str, (usize, usize))>;
```

Comment:
Storing path params in a `Vec` turns out to be faster (and probably safer) than using a `HashMap`. We do the same for query parameters, so maintaining a similar internal API for path params is an added benefit.